sdk/src/builder.rs

```diff
@@ -432,6 +447,29 @@ impl Builder {
                 //println!("adding resource {}", id);
                 builder.resources.add(id, data)?;
             }
+
+            // Load the c2pa_manifests.
+            // We add the manifest data to any ingredient that has a matching active_manfiest label.
+            if file.name().starts_with("manifests/") && file.name() != "manifests/" {
+                let mut data = Vec::new();
+                file.read_to_end(&mut data)?;
+                let manifest_label = file
+                    .name()
+                    .split('/')
+                    .nth(1)
+                    .ok_or(Error::BadParam("Invalid manifest path".to_string()))?;
+                let manifest_label = manifest_label.replace(['_'], ":");
+                for ingredient in builder.definition.ingredients.iter_mut() {
+                    if let Some(active_manifest) = ingredient.active_manifest() {
+                        if manifest_label.starts_with(active_manifest) {
+                            ingredient.set_manifest_data(data.clone())?;
```

Comment:
Can multiple ingredients share the same internal binary manifest? If that's the case, then we need to check if the binary manifest already exists when we write them (ZIP will error if it already exists, check out #499). Otherwise, we can avoid a clone by breaking out of the loop here.

Reply:
Yes, we can have multiple ingredients sharing the same manifest data, so we need to handle that case. It would be nice if Rust had a way to clone only when necessary here.